tron/config/config_parse.py

```diff
@@ -67,9 +67,13 @@ def validator(value, config_context):
         )
 
         try:
-            value % context
+            render_command = value % context
+            # This checks wrong format like "$(A) $(A)s"
+            if render_command != (render_command % context):
```

Comment:
I think this might break commands that escape the string formatting. For example, if the command has `%%Y/%%m/%%d` in it, if you try to format it a second time, it will raise a ValueError.

This is a really tricky situation to catch... "The %(one) something %(stars)s" doesn't raise an error if you format it twice because `something` happens to start with s, which is a valid format type.

Maybe it's enough to detect if rendering the command will cause an error, as we do now. That will prevent people from breaking deployments, at least. But then we can add another feature later that allows you to preview your commands, so that people can manually check that their commands look correct?

Reply:
I agree that we can't "do it twice". @chlgit why isn't the first pass sufficient to detect the bad mapping?

@qui omg you are right:
```
>>> "hello world %(foo)s something" % {'foo': foo}
'hello world hi something'
>>> "hello world %(foo) something" % {'foo': foo}
'hello world hiomething'
```

Reply:
OK, I rewrite the check function. It's not very elegant, but it works.....I hope.

Reply:
@solarkennedy  the first pass can't detect something like this:
 >>> "hello world %(foo) %(foo)s something" % {'foo': foo}
'hello world %(foo)s something'

Reply:
Ah. Yuck :(

We try so hard to validate this kind of easy mistake, users can easily make this mistake all the time (I have), and there are just so many gotchas.

Wouldn't it be cool if we could just snap our fingers and move to `f{}` format strings? Could here possibly be a migration path to this, given we control all our own code and configs?